src/api/app/components/bs_request_state_badge_component.rb

```diff
@@ -9,8 +9,40 @@ def initialize(bs_request:, css_class: nil)
   def call
     content_tag(
       :span,
-      tag.i(class: 'fas fa-code-pull-request me-1').concat(@bs_request.state),
-      class: ['badge', "bg-#{helpers.request_badge_color(@bs_request.state)}", @css_class]
+      tag.i(class: "fas fa-#{decode_state_icon(@bs_request.state)} me-1").concat(@bs_request.state),
+      class: ['badge', "bg-#{decode_state_color(@bs_request.state)}", @css_class]
     )
   end
+
+  def decode_state_color(state)
+    case state
+    when :review, :new
+      'secondary'
+    when :declined
+      'danger'
+    when :superseded
+      'warning'
+    when :accepted
+      'success'
+    when :revoked
+      'dismissed'
+    else
+      'dark'
+    end
+  end
+
+  def decode_state_icon(state)
+    case state
+    when :new
+      'code-branch'
+    when :review, :declined, :revoked
```

Comment:
@ncounter for a decline review we show the `fa-times` icon in other places 
![image](https://user-images.githubusercontent.com/22001671/231167674-028f392c-1412-40f9-8774-0ff94a801231.png)

the `code-pull-request` indicates to me it is still open somehow. Maybe we can use the cross as well?

Reply:
Yeah, @krauselukas I agree it is not the best fit the `code-pull-request`. Problem is, we don't have the `code-pull-request-closed` icon in our font-awesome set because it is not in the _free_ one. So either we stay with some `pull-request`-ish icon, or we use the `fa-times` only. I'd like to avoid adding a custom icon in a different set.
I'd be fine with the `fa-times` only, even if it is no longer clear it is about a code-request by itself. What do you think? Is there any other [font-awesome](https://fontawesome.com/search?o=r&m=free) icon we could use for this special case?

Reply:
@ncounter, I was thinking the same as @krauselukas. And not sure if adding the icon + text in this case make sense or is better to only have the icon or just the text. It would make sense to use both if you have the intention to use them without the "text" in some places.

I will use Github as example:
![Screenshot from 2023-04-11 15-02-09](https://user-images.githubusercontent.com/1212806/231170838-0dae4237-0159-40d7-8f6d-3099434b0321.png)
![Screenshot from 2023-04-11 15-02-25](https://user-images.githubusercontent.com/1212806/231170850-3c5a7874-8856-476a-a96a-f9a6c9eb1a19.png)

But is my opinion :smile_cat:. BTW, nice work :+1:

Reply:
@ncounter  IMO the `fa-times` would be fine, we use it for a declining review on a request, which is what the declined label represent in the end on the request. Someone declined in the review process. Through a quick search on the font awesome library I couldn't spot anything better...

Reply:
TBH, I was not a big fan of the `fa-times` for this. It recalls a delete action and it's also used for failed build results... so I wouldn't use it for request states too.

Taking into account we don't have better free icons, I'd go with your last proposal, yes. So `code-pull-request` + `slash` for `declined` and `revoked` requests.

Reply:
Agree with @saraycp

Reply:
Great, thank you for your feedback.
I implemented the combination of two icons for both `declined` and `revoked`. 

In the PR description I attached the full set of different badge states.

Feel free to give a final review then, Thanks